runtime/src/main/java/org/corfudb/runtime/BootstrapUtil.java

```diff
@@ -76,20 +94,34 @@ private static void bootstrapLayoutServer(IClientRouter router, Layout layout)
      * @param layout Layout to bootstrap with
      */
     private static void bootstrapManagementServer(IClientRouter router, Layout layout)
-            throws ExecutionException, InterruptedException, AlreadyBootstrappedException {
+            throws TimeoutException, NetworkException, AlreadyBootstrappedException {
         ManagementClient managementClient
                 = new ManagementClient(router, layout.getEpoch(), layout.getClusterId());
 
-        try {
-            CFUtils.getUninterruptibly(managementClient.bootstrapManagement(layout),
-                    AlreadyBootstrappedException.class);
-        } catch (AlreadyBootstrappedException abe) {
-            if (!managementClient.getLayout().get().equals(layout)) {
-                log.error("BootstrapUtil: Management Server {}:{} already bootstrapped with "
-                        + "different layout.", router.getHost(), router.getPort());
-                throw abe;
-            }
-        }
+        CompletableFuture<Void> bootstrapManagementServer = managementClient
+                .bootstrapManagement(layout)
+                .exceptionally(ex -> {
+                    try {
+                        CFUtils.unwrap(ex, AlreadyBootstrappedException.class);
+                    } catch (AlreadyBootstrappedException e) {
+                        log.warn("BootstrapUtil: Management Server {}:{} already bootstrapped.",
+                                router.getHost(), router.getPort());
+                        return false;
+                    }
+                    return true;
+                })
+                .thenCompose(bootstrapSucceeded -> {
+                    if (bootstrapSucceeded) return CompletableFuture.completedFuture(layout);
+                    return managementClient.getLayout();
+                })
+                .thenAccept(retrievedLayout -> {
+                    if (retrievedLayout.equals(layout)) {
+                        return;
+                    }
+                    throw new AlreadyBootstrappedException();
```

Comment:
Add a log information.

Reply:
Actually, if you throw an exception you must NOT add any logs.
Adding more info into an exception is ok, but not adding logs, which causes extra logs that are not needed

Reply:
I made an exception more descriptive